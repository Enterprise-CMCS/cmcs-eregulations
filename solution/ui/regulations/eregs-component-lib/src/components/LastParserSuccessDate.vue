<template>
    <span>
        <template v-if="lastParserSuccess">{{ lastParserSuccess }}</template>
        <span v-else class="spinner-span">
            <SimpleSpinner size="xs" />
        </span>
    </span>
</template>

<script>
import { getLastParserSuccessDate } from "utilities/api";
import SimpleSpinner from "./SimpleSpinner.vue";

export default {
    name: "LastParserSuccessDate",

    components: {
        SimpleSpinner,
    },

    props: {
        apiUrl: {
            type: String,
            required: true,
        },
    },

    created() {
        getLastParserSuccessDate({ apiUrl: this.apiUrl })
            .then((response) => {
                this.lastParserSuccess = response;
            })
            .catch(() => {
                this.lastParserSuccess = "N/A";
            });
    },

    data() {
        return {
            lastParserSuccess: "",
        };
    },
};
</script>

<style>
.spinner-span {
    display: inline-block;
    padding: 0 5px;
}
</style>
