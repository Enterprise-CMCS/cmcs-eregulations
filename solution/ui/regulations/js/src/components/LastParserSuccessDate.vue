<template>
    <span>
        <template v-if="lastParserSuccess">{{ lastParserSuccess }}</template>
        <span v-else class="spinner-span">
            <SimpleSpinner size="xs" />
        </span>
    </span>
</template>

<script>
import { getLastParserSuccessDate } from "../../api";
import SimpleSpinner from "./SimpleSpinner.vue";

export default {
    name: "LastParserSuccessDate",

    components: {
        SimpleSpinner,
    },

    props: {
        apiUrl: {
            type: String,
            required: true,
        },
    },

    created() {
        getLastParserSuccessDate(this.apiUrl, { title: 42 })
            .then(
                (response) => {
                    this.lastParserSuccess = response;
                }
            )
            .catch(() => {
                    this.lastParserSuccess = "N/A";
            });
    },

    data() {
        return {
            lastParserSuccess: "",
        };
    },
};
</script>

<style>
.spinner-span {
    display: inline-block;
    padding: 0 5px;
}
</style>
