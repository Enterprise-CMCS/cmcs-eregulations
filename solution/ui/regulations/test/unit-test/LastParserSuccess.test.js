import { render, screen } from "@testing-library/vue";
import { describe, it, expect, beforeEach, afterEach } from "vitest";
import LastParserSuccessDate from "eregsComponentLib/src/components/LastParserSuccessDate.vue";
import flushPromises from "flush-promises";

describe("LastParserSuccessDate", () => {
    beforeEach(() => {});
    afterEach(() => {});
    it("Renders N/A as expected", async () => {
        render(LastParserSuccessDate, {
            props: {
                apiUrl: "test/n/a/",
            },
        });
        await flushPromises();
        const naDate = screen.getByText("N/A");
        expect(naDate).toBeTruthy();
    });
    it("Populates some content", async () => {
        render(LastParserSuccessDate, {
            props: {
                apiUrl: "test/success/",
            },
        });
        await flushPromises();
        const successDate = screen.getByText("Jun 28, 2023");
        expect(successDate).toBeTruthy();
    });
    it("Creates a snapshot of parserdate", async () => {
        const wrapper = render(LastParserSuccessDate, {
            props: {
                apiUrl: "test/snapshot/",
            },
        });
        expect(wrapper).toMatchSnapshot();
    });
});
