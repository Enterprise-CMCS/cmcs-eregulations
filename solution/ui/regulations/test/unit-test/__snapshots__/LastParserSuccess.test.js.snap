// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LastParserSuccessDate > Creates a snapshot of parserdate 1`] = `
{
  "baseElement": <body>
    <div>
      <span>
        <span
          class="spinner-span"
        >
          <div
            class="ds-u-display--flex ds-u-justify-content--center ds-u-align-items--center loading-spinner"
          >
            <span
              class="ds-c-spinner ds-c-spinner--small"
              role="status"
              style="margin: 0px;"
            >
              <span
                class="ds-u-visibility--screen-reader"
              >
                Loading
              </span>
            </span>
          </div>
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <span
        class="spinner-span"
      >
        <div
          class="ds-u-display--flex ds-u-justify-content--center ds-u-align-items--center loading-spinner"
        >
          <span
            class="ds-c-spinner ds-c-spinner--small"
            role="status"
            style="margin: 0px;"
          >
            <span
              class="ds-u-visibility--screen-reader"
            >
              Loading
            </span>
          </span>
        </div>
      </span>
    </span>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
